src/Identity/Controllers/SsoController.cs

```diff
@@ -60,13 +64,16 @@ public async Task<IActionResult> PreValidate(string domainHint)
                 var requestPath = $"/Account/PreValidate?domainHint={domainHint}&culture={culture}";
                 var httpClient = _clientFactory.CreateClient("InternalSso");
                 using var responseMessage = await httpClient.GetAsync(requestPath);
+                var responseJson = await responseMessage.Content.ReadAsStringAsync();
+
                 if (responseMessage.IsSuccessStatusCode)
                 {
-                    // All is good!
-                    return new EmptyResult();
+                    var obj = JsonConvert.DeserializeObject<ExpandoObject>(responseJson, new ExpandoObjectConverter());
+                    return Json(obj);
```

Comment:
Would it be more prudent to just return the JSON string as content instead of deserializing to have it serialized again (since we're not using any of the object properties natively here)?
```suggestion
                    return Content(responseJson, "application/json");
```